### PR TITLE
Google Trusted Publishing docs

### PR DESCRIPTION
**Pull request by di**
Towards #13551, merging adds this to #15192.

(Doesn't actually work right now due to https://github.com/pypi/warehouse/issues/13551#issuecomment-1888051237)

### CODE REVIEW

File: docs/user/trusted-publishers/security-model.md [hunk 1]
```diff
@@ -133,7 +133,21 @@ own security model and considerations.
 
 === "Google Cloud"
 
-    TODO
+    ### Security Model
+
+    If a trusted publisher is configured for a given PyPI project, any service
+    that uses the configured service account can request an OpenID Connect token
+    from Google's identity provider on behalf of that identity. That token can be
+    exchanged for a PyPI API token with the ability to publish to the PyPI project.
```
Comment (facutuesca) [inline docs/user/trusted-publishers/security-model.md:141]: Should we mention here the (optional) possiblity of further restricting the service by configuring the subject ID? Just re-stating what is already mentioned in the `add-a-publisher.md` doc:
```suggestion
    exchanged for a PyPI API token with the ability to publish to the PyPI project.
    The identity used for publishing can be optionally constrained further by
    by specifying the subject, an ID that represents the principal making the request.
```

File: docs/user/trusted-publishers/security-model.md [hunk 2]
```diff
@@ -133,7 +133,23 @@ own security model and considerations.
...
+    The identity used for publishing can be optionally constrained further by
+    by specifying the subject, an ID that represents the principal making the request.
```
Comment (facutuesca) [inline docs/user/trusted-publishers/security-model.md:143]: Whoops, I added an extra `by` :
```suggestion
    The identity used for publishing can be optionally constrained further by
    specifying the subject, an ID that represents the principal making the request.
```

File: docs/user/trusted-publishers/adding-a-publisher.md [hunk 3]
```diff
@@ -45,7 +45,41 @@ each.
...
+
+        Google Cloud also provides [default service
+        accounts](https://cloud.google.com/iam/docs/service-account-types#default)
+        for various products:
+
+          *  Compute Engine: `PROJECT_ID-compute@developer.gserviceaccount.com`
+          *  App Engine: `PROJECT_NAME@appspot.gserviceaccount.com`
+
+        However it is NOT recommended that these be used for publishing, as
+        they are provided by default to every service when they are created.
```
Comment (woodruffw) [inline docs/user/trusted-publishers/adding-a-publisher.md:72]: Nitpick, syntax for bold:

```suggestion
        However it is **not** recommended that these be used for publishing, as
        they are provided by default to every service when they are created.
```

File: docs/user/trusted-publishers/using-a-publisher.md [hunk 4]
```diff
@@ -212,7 +212,56 @@ below describe the setup process for each supported trusted publisher.
...
+    Finally, we can submit that token to PyPI and get a short-lived API token
+    back:
+
+    ```bash
+    resp=$(curl -X POST https://pypi.org/_/oidc/mint-token -d "{\"token\": \"${oidc_token}\"}")
+    api_token=$(jq '.token' <<< "${resp}")
+    ```
+
+    **NOTE**: This is the URL for PyPI. For TestPyPI, the correct
+    domain should be is `test.pypi.org`.
```
Comment (woodruffw) [inline docs/user/trusted-publishers/using-a-publisher.md:258]: Nitpick: callout:

```suggestion
    !!! note
    
        This is the URL for PyPI. For TestPyPI, the correct
        domain should be is `test.pypi.org`.
```

File: docs/user/trusted-publishers/using-a-publisher.md [hunk 5]
```diff
@@ -212,7 +212,56 @@ below describe the setup process for each supported trusted publisher.
...
+    ```
+
+    **NOTE**: `pypi` is only correct for PyPI. For TestPyPI, the correct
+    audience is `testpypi`. More generally, you can access any instance's expected
+    OIDC audience via the `{index}/_/oidc/audience` endpoint:
+
+    ```console
+    $ curl https://pypi.org/_/oidc/audience
+    {"audience":"pypi"}
+    ```
```
Comment (woodruffw) [inline docs/user/trusted-publishers/using-a-publisher.md:247]: Nitpick: callout:

```suggestion
    !!! note
        `pypi` is only correct for PyPI. For TestPyPI, the correct
        audience is `testpypi`. More generally, you can access any instance's expected
        OIDC audience via the `{index}/_/oidc/audience` endpoint:

        ```console
        $ curl https://pypi.org/_/oidc/audience
        {"audience":"pypi"}
        ```
```

**Review by woodruffw — COMMENTED** [review]
LGTM! Three very small formatting nitpicks 🙂

**Review by woodruffw — APPROVED** [review]

File: docs/user/trusted-publishers/using-a-publisher.md [hunk 6]
```diff
@@ -212,7 +212,59 @@ below describe the setup process for each supported trusted publisher.
...
+        $ curl https://pypi.org/_/oidc/audience
+        {"audience":"pypi"}
+        ```
+
+    Finally, we can submit that token to PyPI and get a short-lived API token
+    back:
+
+    ```bash
+    resp=$(curl -X POST https://pypi.org/_/oidc/mint-token -d "{\"token\": \"${oidc_token}\"}")
+    api_token=$(jq '.token' <<< "${resp}")
```
Comment (di) [inline docs/user/trusted-publishers/using-a-publisher.md:255]:
```suggestion
    api_token=$(jq -r '.token' <<< "${resp}")
```
Without this, quotes are introduced into the API token.

File: docs/user/trusted-publishers/using-a-publisher.md [hunk 7]
```diff
@@ -212,7 +212,59 @@ below describe the setup process for each supported trusted publisher.
...
+
+    !!! note
+    
+        This is the URL for PyPI. For TestPyPI, the correct
+        domain should be is `test.pypi.org`.
+
+    This API token can be fed into `twine` or any other uploading client:
+
+    ```bash
+    TWINE_USERNAME=__token__ TWINE_PASSWORD="${api_token}" twine upload dist/*
```
Comment (di) [inline docs/user/trusted-publishers/using-a-publisher.md:266]:
```suggestion
    TWINE_USERNAME=__token__ TWINE_PASSWORD=${api_token} twine upload dist/*
```
Quotes are unnecessary here.

**Comment by di** [comment]
I was able to publish https://test.pypi.org/project/gcp-oidc-test/ from a Google Cloud Compute Engine VM with these instructions:

image

**Pull request merged**